### ISSUE TITLE
should check50 actually !require files?

### ISSUE THREAD

**Issue by dmalan** [cs50/check50#129]
So I ran into a corner case with labs, whereby if a `!require`'d file doesn't exist, then `check50` errs with

```
{'stdout': '', 'stderr': '\x1b[31mYou seem to be missing these required files:\nmario.py\nYou are currently in: /root/sandbox, did you perhaps intend another directory?\x1b[0m\n', 'status': 1, 'timed_out': False, 'success': False}
```

which isn't ideal, since the result can no longer be rendered like any other result. I could just check for `success: False` and render `stderr` as needed, but this particular error isn't ideal, since labs abstract away the `$CWD`, so this hardcoded advice isn't ideal. Thoughts on how best to address?

**Comment by crossroads1112** [comment]
I think `check50` should require files, yes, if for no other reason than to keep the behavior between check50 and submit50 the same. Per the other issue `check50` now spits out:

```json
{
    "error": {
        "type": "MissingFilesError",
        "value": "You seem to be missing these required files:\ncaesar.c\nYou are currently in: /home/chad/Documents/cs50/check50, did you perhaps intend another directory?"
    },
    "version": "3.0.0"
}
```

so you could check the type of the error and display a message accordingly. Does that address your concern?

That does bring up another point though, that by and large our `exists` checks do nothing (other than ensure you don't get a 0 I guess) since they wouldn't even be able to submit if the requisite files don't exist.

**Comment by Jelleas** [comment]
Yup, MissingFilesError is defined here: https://github.com/cs50/lib50/blob/6b94953569791f4ab6b896f8b3e2e42ef64a4edc/lib50/errors.py#L10

`check50.exists` is simply redundant as check50 will only copy those files you specify it to copy. And will fail if the required files are not there before any checks are run. Seems weird to include that into the API. I do like the idea of an exists check, even if just to show some green by default and to confirm that everything is in order (other than just by absence of an error). The `exist` test itself could just be `pass`.

The edge case of students pushing to git themselves is already caught by `lib50`, as `lib50` will throw a `MissingFilesError` (even with `--local` on the remote) before the checks themselves are run (assuming all required files are actually `!require`d).

**Comment by crossroads1112** [comment]
> 

> The edge case of students pushing to git themselves is already caught by lib50, as lib50 will throw a MissingFilesError (even with --local on the remote) before the checks themselves are run (assuming all required files are actually !required).

Ooh that's brings up another interesting bit that if a student submits via git, they won't actually see the exists check fail, they'll see an error.

**Comment by Jelleas** [comment]
Oh whoops, meant `check50.exists(True)` there. Definitely agree `check50.exists` deserves a spot in the api
